lighthouse-core/lib/asset-saver.js

```diff
@@ -229,6 +232,156 @@ function* traceJsonGenerator(traceData) {
   yield '}\n';
 }
 
+/**
+ * @param {LH.Gatherer.Simulation.Result['nodeTimings']} nodeTimings
+ * @return {LH.Trace}
+ */
+function convertNodeTimingsToTrace(nodeTimings) {
+  /** @type {LH.TraceEvent[]} */
+  const traceEvents = [];
+  const baseTs = 1e9;
+  const baseEvent = {pid: 1, tid: 1, cat: 'devtools.timeline'};
+  const frame = 'A00001';
+  /** @param {number} ms */
+  const ts = ms => baseTs + ms * 1000;
+
+  traceEvents.push(createFakeTracingStarted());
+  traceEvents.push({...createFakeTracingStarted(), name: 'TracingStartedInBrowser'});
+
+  // Create a fake requestId counter
+  let requestId = 1;
+  let lastEventEndTime = 0;
+  for (const [node, timing] of nodeTimings.entries()) {
+    lastEventEndTime = Math.max(lastEventEndTime, timing.endTime);
+    if (node.type === 'cpu') {
+      // EvaluateScript
+      const cpuNode = /** @type {LH.Gatherer.Simulation.GraphCPUNode} */ (node);
+      traceEvents.push(createFakeEvaluateScript(cpuNode, timing));
```

Comment:
they're all represented here, we're just using `EvaluateScript` as the single name rather than differentiating since there can be multiple of those bundled together.

added a comment to this effect

Reply:
aiight so our CPU nodes in the dep graph represent scheduler tasks, which are normally not shown in devtools so we can't visualize using those event names without using Show All Events.. So we're just picking a random name even though the work may not be script. I wonder if we can use another event name that's a bit more ambiguous.   i'm thinking Task/Program. it depends on how the flamechart chooses what to visualize but i think there may be something better than painting all tasks yellow.

also.. wdyt about including the cpuNode's childEvents with their time adjustment? i think that'd be pretty powerful. and lantern did consider them so it feels fair.

Reply:
> So we're just picking a random name even though the work may not be script. 

bingo

> I wonder if we can use another event name that's a bit more ambiguous. i'm thinking Task/Program.

sure just lemme know what you pick happy to change 👍, I just want a distinguishable color that I can visualize the toplevel tasks

> wdyt about including the cpuNode's childEvents with their time adjustment? i think that'd be pretty powerful. and lantern did consider them so it feels fair.

sounds great! followup?

Reply:
trace event name: Task
trace event category: (keep it as devtools.timeline)

we get this without any experiments:

![image](https://user-images.githubusercontent.com/39191/41181085-64ccd9ac-6b25-11e8-9122-331839f0fd6a.png)

Reply:
>> wdyt about including the cpuNode's childEvents with their time adjustment? i think that'd be pretty powerful. and lantern did consider them so it feels fair.

> sounds great! followup?

how about in this PR? feels like it's like +7 lines. and it'd be so juicy. :D

Reply:
then we're either thinking of different things or I'm waaaaay worse at this than you cuz I've got a 50-line diff sofar and no useful output yet 😆

Reply:
https://github.com/GoogleChrome/lighthouse/compare/lantern_to_trace...lantern_to_trace_with_cpu_detail